Chewbawka2017/src/org/team1540/chewbawka/GearSlider.java

```diff
@@ -1,7 +1,98 @@
+				"returning to center",
+				"calibrated");
+		
+		gearSliderMotor.modEncoder().configureEncoderCodesPerRev(125 * 15);
+		gearSliderMotor.modGeneralConfig().configureMaximumOutputVoltage(12f, -12f);
+		FloatCell calibratingSpeed = new FloatCell();
+		FloatOutput gearSliderSpeedControl = gearSliderMotor.asMode(OutputControlMode.SPEED_FIXED);
+		FloatOutput gearSliderPositionControl = gearSliderMotor.asMode(OutputControlMode.POSITION_FIXED);
+		FloatIO gearSliderPosition = gearSliderMotor.modEncoder().getEncoderPosition();
+		FloatInput gearSliderVelocity = gearSliderMotor.modEncoder().getEncoderVelocity();
+		
+		FloatCell calibrationSpeed = new FloatCell(10f);
+		FloatCell slow = new FloatCell(2f);
+		BooleanInput tooSlow = gearSliderVelocity.absolute().atMost(slow);
+		FloatCell slidingDistance = new FloatCell();
+		BooleanCell calibrated = new BooleanCell();
+		FloatCell positionError = new FloatCell(10f);
+		BooleanInput atMiddle = gearSliderPosition.inRange(slidingDistance.dividedBy(2f).minus(positionError), 
+				slidingDistance.dividedBy(2f).plus(positionError));
+		EventOutput resetEncoder = gearSliderPosition.eventSet(0f);
+		EventOutput recordDistance = slidingDistance.eventSet(gearSliderPosition);
+		
+		EventCell calibrate = new EventCell();
+		gearSliderStates.setStateWhen("calibrating 0 end", calibrate);
+		gearSliderStates.setStateWhen("calibrating distance", 
+				tooSlow.onPress().and(gearSliderStates.getIsState("calibrating 0 end")));
+		gearSliderStates.setStateWhen("returning to center", 
+				tooSlow.onPress().and(gearSliderStates.getIsState("calibrating distance")));
+		gearSliderStates.setStateWhen("calibrated", 
+				atMiddle.onPress().and(gearSliderStates.getIsState("returning to center")));
+		
+		calibrate.on(FRC.startTele);
+		gearSliderSpeedControl.setWhen(calibrationSpeed.negated(), 
+				gearSliderStates.onEnterState("calibrating 0 end").or(gearSliderStates.onEnterState("returning to center")));
+		gearSliderSpeedControl.setWhen(calibrationSpeed, gearSliderStates.onEnterState("calibrating distance"));
+		gearSliderSpeedControl.setWhen(0f, gearSliderStates.onEnterState("calibrated"));
+		resetEncoder.on(gearSliderStates.onExitState("calibrating 0"));
+		recordDistance.on(gearSliderStates.onExitState("calibrating distance"));
+		calibrated.setWhen(true, gearSliderStates.onExitState("calibrating distance"));
+		
+		BooleanInput sliderTooFarF = gearSliderPosition.atLeast(slidingDistance);
+		BooleanInput sliderTooFarB = gearSliderPosition.atMost(0f);
+		sliderTooFarF.onPress(gearSliderPositionControl.eventSet(slidingDistance));
+		sliderTooFarB.onPress(gearSliderPositionControl.eventSet(0f));
+		FloatCell slidingControlScaling = new FloatCell(1f);
+		ControlBindings.gearSliderControls.onChange().and(calibrated).send(gearSliderPositionControl.eventSet(
+				gearSliderPosition.plus(ControlBindings.gearSliderControls.multipliedBy(slidingControlScaling))));
+		
+		Cluck.publish("Gear Servo Left Output", servoLeft);
```

Comment:
Given this class isn't static, this will be funny if multiple instances of the class are constructed. Either add a name field to the constructor and use that in Cluck.publish(), make the class static, or stop publishing these values.